File: code/backend/java/core/src/main/java/games/paths/core/model/story/CatalogExportResult.java
package games.paths.core.model.story;

import java.util.List;

/**
 * CatalogExportResult - v0.37.6 outcome of a static catalog export: where the
 * files went and one entry per written language.
 */
public record CatalogExportResult(String target, List<CatalogFile> files) {

    /** One written catalog file. */
    public record CatalogFile(String lang, String path, int count, int bytes) {}
}

File: code/backend/python/app/core/models/story/catalog_export_result.py
from dataclasses import dataclass, field
from typing import List


@dataclass
class CatalogFile:
    lang: str
    path: str
    count: int
    bytes: int


@dataclass
class CatalogExportResult:
    """v0.37.6 — outcome of a static catalog export: destination + one entry per language."""
    target: str
    files: List[CatalogFile] = field(default_factory=list)
